pkg/install/v1alpha1/artifact.go

```diff
@@ -64,53 +63,40 @@ func IsNotRunTask(given *Artifact) bool {
 	return given != nil && !strings.Contains(given.Doc, string(RunTaskArtifact))
 }
 
+func (a *Artifact) Template(values map[string]interface{}, t template.Templater) (u *Artifact, err error) {
+	if a == nil {
```

Comment:
should we add this type of checks to other methods where the pointer is getting used?